packages/core/src/util/transactionSummaryInspector.ts

```diff
@@ -111,14 +115,38 @@ const getUnaccountedFunds = async (
   return subtractValueQuantities([totalOutputs, totalInputs]);
 };
 
+const toAssetInfoWithAmount = async (
+  assetProvider: AssetProvider,
+  tokenMap?: TokenMap
+): Promise<Map<Cardano.AssetId, AssetInfoWithAmount>> => {
+  if (!tokenMap) return new Map();
+
+  const assetIds = uniq(tokenMap && tokenMap.size > 0 ? [...tokenMap.keys()] : []);
```

Comment:
A `Map` should already have uniq keys :thinking: 
```suggestion
  const assetIds =tokenMap && tokenMap.size > 0 ? [...tokenMap.keys()] : [];
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1076/commits/2ee48609f1598d9c98ec1a235d796302a0c5b4e6